### PR TITLE
Frogger-api-task

### PR DESCRIPTION
**Pull request by Conversee12**
### frogger-api-task

[Demo](https://conversee12.github.io/frontend-nanodegree-arcade-game/) |
[Code base](https://github.com/Conversee12/frontend-nanodegree-arcade-game.git)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Comment by Conversee12** [comment]
I've completed self-checks, did'nt find any common errors

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@Conversee12 good job. Although further improvements are required.
The comments below refer items from the [requirements](https://kottans.org/documentation/docs/doc/code-review/#oo-js-frogger)

File: submissions/Conversee12/Frogger-api-task/app.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    this.x += this.speed * dt;
+    if (this.x > End_X) {
+        this.x = Start_X;
+        this.speed = speedLevel + Math.floor(Math.random() * multiplierSpeed)
+    }
+
+    if (this.x < player.x + PlayerWidth && this.y < player.y + PlayerHeight && this.x + PlayerWidth > player.x && this.y + PlayerHeight > player.y) {
+        setTimeout(function () {
+            player.x = InitialPosition_X;
+            player.y = InitialPosition_Y;
```
Comment (OleksiyRudenko) [inline submissions/Conversee12/Frogger-api-task/app.js:37]: `player` is a global variable. 4.iii provides a hint on how to fix the issue.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@Conversee12 
Feel free asking questions if anything is not clear

**Comment by Conversee12** [comment]
image
Is this "allEnemies" a global variable? (if yes) Honestly, i don't know how to change this. Please, explain.

**Comment by OleksiyRudenko** [comment]
> image Is this "allEnemies" a global variable? (if yes) Honestly, i don't know how to change this. Please, explain.

Original requirement referred specifically to classes definitions, but yes, writing clean functions (functions that use only what is passed to them explicitly via parameters) is a good practice. What blocks you to re-use your experience with global `player`?

You do not need to fix this.

**Comment by OleksiyRudenko** [comment]
```
setTimeout(function () {
            player.x = InitialPosition_X;
            player.y = InitialPosition_Y;
})
```

Check what makes arrow functions different from regular functions. `setTimeout` is a classic example for this.

**Comment by Conversee12** [comment]
> > image Is this "allEnemies" a global variable? (if yes) Honestly, i don't know how to change this. Please, explain.
> 
> Original requirement referred specifically to classes definitions, but yes, writing clean functions (functions that use only what is passed to them explicitly via parameters) is a good practice. What blocks you to re-use your experience with global `player`?
> 
> You do not need to fix this.

Something went wrong, I'll try again to figure it out myself

**Comment by Conversee12** [comment]
> ```
> setTimeout(function () {
>             player.x = InitialPosition_X;
>             player.y = InitialPosition_Y;
> })
> ```
> 
> Check what makes arrow functions different from regular functions. `setTimeout` is a classic example where arrow functions really help.

Read an article about this; understood and corrected

**Review by OleksiyRudenko — APPROVED** [review]
@Conversee12 great job!

**Pull request merged**